snmplib/mib.c

```diff
@@ -647,9 +647,15 @@ sprint_realloc_octet_string(u_char ** buf, size_t * buf_len,
     case NETSNMP_STRING_OUTPUT_GUESS:
         hex = 0;
         for (cp = var->val.string, x = 0; x < (int) var->val_len; x++, cp++) {
+#if defined(_MSC_VER) /* Windows MSVC library includes non-ascii characters (above 0x7F) as printing, which can generate ugly strings */
```

Comment:
One of the goals of the Net-SNMP project is to have the same behavior the same on all supported platforms. So I'm less than enthusiast about adding `#if defined(_MSC_VER)` here.